### ISSUE TITLE
Possible false positive for redundant_closure_call

### ISSUE THREAD

**Issue by nico-abram** [rust-lang/rust-clippy#4705]
Playground link: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=fd3adf287da8d4e6ee17a97aec2ec8be

Code:

```rs
#[allow(clippy::identity_op)]
#[inline]
fn fill_output_color(output_texture: &mut [u8], idx: usize, color: u16) {
    let r = ((color >> 0) & 0x001F) as u8;
    let g = ((color >> 5) & 0x001F) as u8;
    let b = ((color >> 10) & 0x001F) as u8;

    let col5_to_col8 = |x| (x << 3) | (x >> 2);
    output_texture[(idx * 3) + 0] = col5_to_col8(r); // Lint triggered here
    output_texture[(idx * 3) + 1] = col5_to_col8(g);
    output_texture[(idx * 3) + 2] = col5_to_col8(b);
}
```

**Comment by mikerite** [comment]
Is this a duplicate of #3354?

**Comment by flip1995** [comment]
Duplicate of #3354 

Thanks for noticing @mikerite
